# [ACCOUNTING] post-dbNSFP filtering and local download

## confirming that non-detected & detected CK positions are correct/ready to be made into .bed file for VEP processing!

12.10.19

### Brief reminder of dbNSFP filtering from searching old html files in 'MAPPING/HTML_before_clean/': 
- only kept rows where stable canonical uniprot ID was found (searched 3903...2 were chr multimapping so total 3901)
    - 2 uniprot IDs were mapping to multiple chromosomes (error in bed file from uniprot)
- lost rows that did not have valid hg19 coordinates, failed CADDhg19 annotation (2165 rows)
- P42166 & P42167 are isoforms on chr12 that caused issues in dbnsfp, ID was combined for some rows because target position found in both IDs..i think i made combo rows into 'P42166'
- BUG FIX entailed searching for canonical uniprot ID in dbNSFP col list, then using index, comparing AApos in list based on index
    - result was equivalent AA position for found canonical uniprot ID!
- final cleaning of merged dbNSFP (all chr merge), filtered out non missense based on aagain column like 'x', also AA positions with negative (i.e. C-1) splice variants

## in K markdown:
1. check that all uniprot IDs between detected and non detected files match
2. check that all targets are found in subset of IDs
3. check that all nontarget CK positions are accounted for by comparing to uniprot fasta raw CK totals
4. check that all C and K codons have 7 nucleotide changes overlapping...allowed 2 positions to not have 7, both are nontarget positions..notes below. 
5. make bed files

```python
# from non-detected file, more detail at end of markdown

    pos_ID	codon_changes(7)
    Q86W56_K115	  5
    Q9Y6J0_C571	  6
    
```

### FINAL ACCOUNTING of filtered/checked CK positions for .bed file:
* total uniprot IDs = 3840
* all targets found with 7 changes per codon
* only 2 CK non-targets missing and 2 aminos with incorrect codon change count

| TYPE | Cysteine | Lysine | 
|-------|--------------|-----------|
| Low | 895 | 3401 | 
| Medium | 412 | 660 |
| High | 94 | 302 |
| Panreactive | 5088 | 7804 |
| TargetSM | 708 | 115 |
| EVER-labeled | 6057 | 8868 |
| not-labeled | 34050 | 140652 |

# [1] check that detected and detected files with coordinates of Cysteine and Lysine have the same unique uniprot IDs

In [7]:
# packages
import os
import sys
import numpy as np
import pandas as pd
from statistics import stdev 
from ast import literal_eval # for mismap_score func
from statistics import mean

In [8]:
from IPython.display import display, HTML
pd.set_option('display.max_columns', None)
pd.options.display.max_seq_items = 2000
display(HTML("<style>.container {width:90% !important;}</style>"))
sys.path.append("/Users/mariapalafox/Desktop/Toolbox")
from all_funx import *

In [9]:
os.chdir("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/CADDmapped/BUG_FIX")

In [10]:
det = pd.read_csv("detected_appeneded_merge.csv", low_memory=False)
notd = pd.read_csv("not_detected_appended.csv", low_memory=False)

In [13]:
det.head(3)

pos_id19        pos_id38_x aaref_x aaalt_x  matched_aapos_x  \
0  10_000093270_A_C  10_000047330_A_C       C       W              354   
1  10_000093271_C_A  10_000047331_C_A       C       F              354   
2  10_000093271_C_G  10_000047331_C_G       C       S              354   

   CADD_phred_hg38 matched_UKBID_x                                 pos_dict  \
0             22.6          Q3ZCM7  {12: 'C', 239: 'C', 303: 'C', 354: 'C'}   
1             22.6          Q3ZCM7  {12: 'C', 239: 'C', 303: 'C', 354: 'C'}   
2             22.4          Q3ZCM7  {12: 'C', 239: 'C', 303: 'C', 354: 'C'}   

  matched_target  CADD_phred_hg19  CADDdiff_38minus19 Amino_acids  \
0           C354            22.30                0.30     Cys/Trp   
1           C354            21.80                0.80     Cys/Phe   
2           C354            19.67                2.73     Cys/Ser   

        pos_ID  Cys_reactivity Cys_react_threshold Cys_target_label  \
0  Q3ZCM7_C354            5.37                 Low              NaN   
1  Q3ZCM7_C354            5.37                 Low              NaN   
2  Q3ZCM7_C354            5.37                 Low              NaN   

   Lys_reactivity Lys_react_threshold Lys_target_label #chr  pos(1-based) ref  \
0             NaN                 NaN              NaN   10         47330   A   
1             NaN                 NaN              NaN   10         47331   C   
2             NaN                 NaN              NaN   10         47331   C   

  alt aaref_y aaalt_y rs_dbSNP151 hg19_chr  hg19_pos(1-based) hg18_chr  \
0   C       C       W           .       10            93270.0       10   
1   A       C       F           .       10            93271.0       10   
2   G       C       S           .       10            93271.0       10   

  hg18_pos(1-based)        aapos genename  \
0             83270  320;317;354    TUBB8   
1             83271  320;317;354    TUBB8   
2             83271  320;317;354    TUBB8   

                                    Ensembl_geneid  \
0  ENSG00000261456;ENSG00000261456;ENSG00000261456   
1  ENSG00000261456;ENSG00000261456;ENSG00000261456   
2  ENSG00000261456;ENSG00000261456;ENSG00000261456   

                              Ensembl_transcriptid  \
0  ENST00000564130;ENST00000568866;ENST00000568584   
1  ENST00000564130;ENST00000568866;ENST00000568584   
2  ENST00000564130;ENST00000568866;ENST00000568584   

                                 Ensembl_proteinid               Uniprot_acc  \
0  ENSP00000457610;ENSP00000457062;ENSP00000456206  Q5SQY0;A0A075B736;Q3ZCM7   
1  ENSP00000457610;ENSP00000457062;ENSP00000456206  Q5SQY0;A0A075B736;Q3ZCM7   
2  ENSP00000457610;ENSP00000457062;ENSP00000456206  Q5SQY0;A0A075B736;Q3ZCM7   

                              Uniprot_entry                HGVSc_ANNOVAR  \
0  Q5SQY0_HUMAN;A0A075B736_HUMAN;TBB8_HUMAN  c.960T>G;c.951T>G;c.1062T>G   
1  Q5SQY0_HUMAN;A0A075B736_HUMAN;TBB8_HUMAN  c.959G>T;c.950G>T;c.1061G>T   
2  Q5SQY0_HUMAN;A0A075B736_HUMAN;TBB8_HUMAN  c.959G>C;c.950G>C;c.1061G>C   

             HGVSp_ANNOVAR                 HGVSc_snpEff  \
0  p.C320W;p.C317W;p.C354W  c.960T>G;c.951T>G;c.1062T>G   
1  p.C320F;p.C317F;p.C354F  c.959G>T;c.950G>T;c.1061G>T   
2  p.C320S;p.C317S;p.C354S  c.959G>C;c.950G>C;c.1061G>C   

                          HGVSp_snpEff                    HGVSc_VEP  \
0  p.Cys320Trp;p.Cys317Trp;p.Cys354Trp  c.960T>G;c.951T>G;c.1062T>G   
1  p.Cys320Phe;p.Cys317Phe;p.Cys354Phe  c.959G>T;c.950G>T;c.1061G>T   
2  p.Cys320Ser;p.Cys317Ser;p.Cys354Ser  c.959G>C;c.950G>C;c.1061G>C   

                             HGVSp_VEP          APPRIS GENCODE_basic    TSL  \
0  p.Cys320Trp;p.Cys317Trp;p.Cys354Trp  .;.;principal1         Y;Y;Y  5;5;1   
1  p.Cys320Phe;p.Cys317Phe;p.Cys354Phe  .;.;principal1         Y;Y;Y  5;5;1   
2  p.Cys320Ser;p.Cys317Ser;p.Cys354Ser  .;.;principal1         Y;Y;Y  5;5;1   

  VEP_canonical cds_strand refcodon  codonpos  codon_degeneracy  \
0       .;.;YES          -      TGT         3                 2   

In [15]:
notd.head(3)

pos_id19          pos_id38 aaref aaalt  matched_aapos  \
0  10_000093156_C_A  10_000047216_C_A     K     N            392   
1  10_000093156_C_G  10_000047216_C_G     K     N            392   
2  10_000093157_T_A  10_000047217_T_A     K     M            392   

                             HGVSp_VEP  CADD_phred_hg38 matched_UKBID  \
0  p.Lys358Asn;p.Lys355Asn;p.Lys392Asn             23.0        Q3ZCM7   
1  p.Lys358Asn;p.Lys355Asn;p.Lys392Asn             23.1        Q3ZCM7   
2  p.Lys358Met;p.Lys355Met;p.Lys392Met             23.2        Q3ZCM7   

                                  pos_dict  matched_target  CADD_phred_hg19  \
0  {12: 'C', 239: 'C', 303: 'C', 354: 'C'}           False             22.9   
1  {12: 'C', 239: 'C', 303: 'C', 354: 'C'}           False             22.8   
2  {12: 'C', 239: 'C', 303: 'C', 354: 'C'}           False             23.2   

   CADDdiff_38minus19 lost_amino gained_amino Amino_acids pos_ID_falseCKtarget  
0                 0.1        Lys          Asn     Lys/Asn          Q3ZCM7_K392  
1                 0.3        Lys          Asn     Lys/Asn          Q3ZCM7_K392  
2                 0.0        Lys          Met     Lys/Met          Q3ZCM7_K392

In [35]:
# DETECTED FILE
# how many unique uniprot IDs: 
uniqueCount(det, 'matched_UKBID_x')
# how many unique c and k positions
uniqueCount(det, 'pos_ID')

# not detected file
uniqueCount(notd, 'matched_UKBID')
uniqueCount(notd, 'pos_ID_falseCKtarget')

len of col:  106064
len of col set:  3880

len of col:  106064
len of col set:  15152

len of col:  1259801
len of col set:  3878

len of col:  1259801
len of col set:  179957



In [41]:
# controlling for same uniprot IDs in both detected and non detected files
detID = list(det.matched_UKBID_x)
notdID = list(notd.matched_UKBID)

difflist = diffList(detID, notdID)
intersec = intersectionCheck(detID, notdID)

print(difflist)
print(len(intersec))
# result of detectecd - notdetected, so these IDs are in the det set only

'''
['Q9Y4H4', 'P63220', 'O95777', 'P62857']
3876
'''
intersec = list(intersec)
print(len(intersec))

different number of IDs :  4
['Q9Y4H4', 'P63220', 'O95777', 'P62857']
3876
3876


In [52]:
# combining det and not det CK positions and entries to compare to uniprot fasta file
sim_det = det[['matched_UKBID_x', 'pos_ID']]
sim_not = notd[['matched_UKBID', 'pos_ID_falseCKtarget']]
# rename col
sim_det.columns = ['entry', 'pos_ID']
sim_not.columns = ['entry', 'pos_ID']
# combine 
dbnsfp_ck = pd.concat([sim_det, sim_not])
dbnsfp_ck.drop_duplicates(inplace=True)

# drop rows with IDs not in shared 3876 set
dbnsfp_3876 = addcolumnconditionalDrop(intersec, dbnsfp_ck, 'entry', 'shared3876')
detected3876 = addcolumnconditionalDrop(intersec, sim_det, 'entry', 'shared3876')

  shared3876   Count
0       True  195046
1      False      63
dropping false
df shape post drop:  (195046, 3)
  shared3876   Count
0       True  105959
1      False     105
dropping false
df shape post drop:  (105959, 3)
(195046, 3)
(105959, 3)


/Users/mariapalafox/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:376: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[key] = _infer_fill_value(value)
/Users/mariapalafox/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:494: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[item] = s
/Users/mariapalafox/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:4102: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the cav

In [56]:
print(len(detected3876.entry.unique()))
detected3876.drop_duplicates(inplace=True)
print(detected3876.head(5))
# ONLY DETECTED POSITIONS IN DBNSFP

3876
     entry       pos_ID shared3876
0   Q3ZCM7  Q3ZCM7_C354       True
7   Q3ZCM7  Q3ZCM7_C303       True
14  Q3ZCM7  Q3ZCM7_C239       True
21  Q3ZCM7   Q3ZCM7_C12       True
28  Q92615  Q92615_C633       True


/Users/mariapalafox/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [75]:
# grouping dbnsfp detected file only by uniprot ID
detected2 = detected3876.groupby('entry')['pos_ID'].apply(lambda x: len(x))

In [76]:
# from chemoproteomics data processing:
targetcounts = pd.read_csv("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/CYS_LYS_2012_to2018UKBpositions/REFERENCE_2018_mappedCK/SUBSETS/detection_freq_3953_CK_all.csv")
c_target = pd.read_csv("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/CYS_LYS_2012_to2018UKBpositions/REFERENCE_2018_mappedCK/SUBSETS/Cys2018_ever_labeled_6234.csv")
k_target = pd.read_csv("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/CYS_LYS_2012_to2018UKBpositions/REFERENCE_2018_mappedCK/SUBSETS/Lys2018_ever_labeled_9042.csv")

In [77]:
detected2 = pd.DataFrame(detected2)
detected2.reset_index(inplace=True)
detected2.columns = ['ID', 'dbNSFP_target_count']
detected2.set_index('ID', inplace=True)
targetcounts.sort_values('ID', inplace=True)
targetcounts.set_index('ID', inplace=True)

dbNSFP_target_count
ID                         
A0AVT1                   11
A0JNW5                    1
A0MZ66                    1
A1KXE4                    1
A1L0T0                    1
...                     ...
Q9Y6M9                    1
Q9Y6W3                    2
Q9Y6W5                    1
Q9Y6W6                    1
Q9Y6Y8                    1

[3876 rows x 1 columns]

In [67]:
detected2.reset_index(inplace=True)
targetcounts.reset_index(inplace=True)
mergetargets = pd.merge(targetcounts,detected2,how='inner',on=['ID'])
mergetargets

ID  times_detected  count_C_targets  count_K_targets  \
0     A0AVT1              11                6                5   
1     A0JNW5               1                0                1   
2     A0MZ66               1                1                0   
3     A1KXE4               1                1                0   
4     A1L0T0               1                0                1   
...      ...             ...              ...              ...   
3837  Q9Y6M9               1                1                0   
3838  Q9Y6W3               2                2                0   
3839  Q9Y6W5               1                1                0   
3840  Q9Y6W6               1                1                0   
3841  Q9Y6Y8               1                1                0   

      dbNSFP_target_count  
0                      11  
1                       1  
2                       1  
3                       1  
4                       1  
...                   ...  
3837                    1  
3838                    2  
3839                    1  
3840                    1  
3841                    1  

[3842 rows x 5 columns]

In [68]:
mergetargets['check_target_total'] = mergetargets['times_detected'] - mergetargets['dbNSFP_target_count']
checkColumnValues(mergetargets, 'check_target_total')

   check_target_total  Count
0                   0   3841
1                   1      1


In [70]:
# how many IDs are in dbNSFP detected file and not in newly filtered step 1 of mapping data (post 2012 positions mapping)
targetlist = targetcounts.ID.tolist()
detectedlist = detected2.ID.tolist()
# checking list of IDs not found in chemoproteomic filtered files
difflist = diffList(detectedlist, targetlist)
difflist

different number of IDs :  34


['P54136',
 'P86791',
 'P68371',
 'Q71DI3',
 'Q9Y2S7',
 'O75044',
 'P58107',
 'Q9NXV6',
 'Q99873',
 'P63261',
 'P62826',
 'Q8WUH1',
 'Q4G0P3',
 'P0DMV9',
 'Q8WUH6',
 'P82921',
 'P19105',
 'P06753',
 'Q8IYB9',
 'P04406',
 'Q9BTY7',
 'P68363',
 'Q9BQE3',
 'P80303',
 'Q9GZM3',
 'Q99613',
 'P60174',
 'Q96TA1',
 'P08238',
 'Q86YV5',
 'P04350',
 'P26640',
 'Q96GY3',
 'P63241']

## why are some uniprot IDs in the dbNSFP detected file and not in uniprot chemoproteomics ever labeled files??

**these uniprot IDs were dropped when i corrected the CK position checked script for 2012 chemoposition mapping to 2018 reference sequences**
- only keep IDs where all detected positions found (n=3841)...final count 3840

# [2] check that detected and detected files have all C and K positions for uniprot ID 

In [74]:
all_targets = mergetargets[mergetargets['check_target_total'] != 1].copy()
all_targets

ID  times_detected  count_C_targets  count_K_targets  \
0     A0AVT1              11                6                5   
1     A0JNW5               1                0                1   
2     A0MZ66               1                1                0   
3     A1KXE4               1                1                0   
4     A1L0T0               1                0                1   
...      ...             ...              ...              ...   
3837  Q9Y6M9               1                1                0   
3838  Q9Y6W3               2                2                0   
3839  Q9Y6W5               1                1                0   
3840  Q9Y6W6               1                1                0   
3841  Q9Y6Y8               1                1                0   

      dbNSFP_target_count  check_target_total  
0                      11                   0  
1                       1                   0  
2                       1                   0  
3                       1                   0  
4                       1                   0  
...                   ...                 ...  
3837                    1                   0  
3838                    2                   0  
3839                    1                   0  
3840                    1                   0  
3841                    1                   0  

[3841 rows x 6 columns]

In [84]:
all_targets.sum()

ID                     A0AVT1A0JNW5A0MZ66A1KXE4A1L0T0A2A288A2RUC4A3KN...
times_detected                                                     14926
count_C_targets                                                     6058
count_K_targets                                                     8868
dbNSFP_target_count                                                14926
check_target_total                                                     0
dtype: object

In [118]:
# dropped id found with weird K codon missense counts and checking again post drop
# drop rows with IDs not in shared 3876 set
all3840 = all_targets.ID.tolist()
all3840.remove('Q8WZ42')
print(len(all3840))

det_final = addcolumnconditionalDrop(all3840, det, 'matched_UKBID_x', 'shared3841')
not_final = addcolumnconditionalDrop(all3840, notd, 'matched_UKBID', 'shared3841')

  shared3841   Count
0       True  104475
1      False       7
dropping false
df shape post drop:  (104475, 400)
  shared3841    Count
0       True  1222911
1      False    24297
dropping false
df shape post drop:  (1222911, 17)
len of col:  104475
len of col set:  3840

len of col:  104475
len of col set:  14925

len of col:  1222911
len of col set:  3840

len of col:  1222911
len of col set:  174702



In [117]:
all_targets.to_csv("all_targetCK_found_in_UKBIDs3840_dbNSFP.csv")

## [3] checking that all missense for C and K codon have 7 reported alterations
- all target positions had 7 changes for codon (correct)
- non targets had 3 IDs with not 7 count for codon changes, dropped worst ID

In [119]:
# checking that all missense and in file
det_final_group = det_final.groupby('pos_ID')['pos_id19'].apply(lambda x:len(x))
not_final_group = not_final.groupby('pos_ID_falseCKtarget')['pos_id19'].apply(lambda x:len(x))

det_final_group = pd.DataFrame(det_final_group)
checkColumnValues(det_final_group, 'pos_id19') # all target codons had 7

not_final_group = pd.DataFrame(not_final_group)
checkColumnValues(not_final_group, 'pos_id19')
'''
BEFORE DROPPING TTN gene
pos_id19   Count
0         7  178139
1        14      16
2         6       1
3         5       1
# detected file has 3 uniprot IDs with amino acids containing incorrect missense codon count
'''

   pos_id19  Count
0         7  14925
   pos_id19   Count
0         7  174700
1         6       1
2         5       1


'\nBEFORE DROPPING TTN gene\npos_id19   Count\n0         7  178139\n1        14      16\n2         6       1\n3         5       1\n# detected file has 3 uniprot IDs with amino acids containing incorrect missense codon count\n'

In [124]:
# BEFORE DROPPING Q8WZ42 ID
#INCORRECT CODON NUMBER IN NOT detected file from dbNSFP
not_final_group[not_final_group['pos_id19'] != 7]

pos_id19
pos_ID_falseCKtarget          
Q86W56_K115                  5
Q9Y6J0_C571                  6

## keeping: 
Q9Y6J0 has 2 targets is Calcineurin-binding protein cabin-1 important for T cell receptor signaling

Q86W56 has 3 target is PolyADP-ribose glycohydrolase with PCNA binding domain dont delete!

## dropping: 
Q8WZ42 has 1 target TTN important for straited muscle

```
pos_ID_falseCKtarget	
Q8WZ42_K11429	14
Q8WZ42_K11435	14
Q8WZ42_K11436	14
Q8WZ42_K11443	14
Q8WZ42_K11444	14
Q8WZ42_K11451	14
Q8WZ42_K11457	14
Q8WZ42_K11463	14
Q8WZ42_K11464	14
Q8WZ42_K11471	14
Q8WZ42_K11472	14
Q8WZ42_K11479	14
Q8WZ42_K11485	14
Q8WZ42_K11491	14
Q8WZ42_K11492	14
Q8WZ42_K11500	14
```

# ALL targets in 3840 uniprot IDs found!

In [127]:
# DETECTED FILE 
uniqueCount(det_final, 'matched_UKBID_x')
uniqueCount(det_final, 'pos_ID')
print()
# not detected file
uniqueCount(not_final, 'matched_UKBID')
uniqueCount(not_final, 'pos_ID_falseCKtarget')

len of col:  104475
len of col set:  3840

len of col:  104475
len of col set:  14925


len of col:  1222911
len of col set:  3840

len of col:  1222911
len of col set:  174702



# [4] now checking CK total counts for these IDs (uniprot fasta) vs total CK not detected from dbNSFP
- all CK found exceptp 2 postions, great!

In [134]:
ukb = pd.read_csv("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/CYS_LYS_2012_to2018UKBpositions/REFERENCE_2018_mappedCK/SUBSETS/countsfromfastaCCDS_filter_counts_all_AA.csv")
ukb_final = addcolumnconditionalDrop(all3840, ukb, 'ID', 'shared3840')
ukb_final.sum()

  shared3840  Count
0      False  14592
1       True   3840
dropping false
df shape post drop:  (3840, 30)


Unnamed: 0                                              33973922
ID             Q9HAS0Q86X76Q9NQR4Q9Y314Q04721Q9UHY1P49116Q166...
entryName      NJMU_HUMANNIT1_HUMANNIT2_HUMANNOSIP_HUMANNOTC2...
Length                                                   2276173
proSequence    MLPSLQESMDGDEKELESSEEGGSAEERRLEPPSSSHYCLYSYRGS...
A                                                         164566
B                                                              0
C                                                          40115
D                                                         119895
E                                                         179808
F                                                          77005
G                                                         143874
H                                                          54491
I                                                         100800
K                                                         149550
L                        

## results of non detected CK accounting: 

* Total C = 40,115
* Total K = 149,550
* Sum in ukb fasta = 189,665
* sumUKB - dbNSFPnontarget = 14,963...which is 2 less than total number of target C and K...so only short 2 amino acids from total count!

In [133]:
print(149550 + 40115) # CK sum in ukb
print(189665 - 174702) # difference btw ukb and dbNSFP non detected CK positions
# 14963 , which is 2 less than it is suppose to be

189665
14963


In [147]:
# det_final_accounting = det_final[['matched_UKBID_x', 'pos_ID','Cys_reactivity', 'Cys_react_threshold', 'Cys_target_label', 'Lys_reactivity', 'Lys_react_threshold', 'Lys_target_label']]
det_final_accounting = det_final[['aaref_x', 'pos_ID']]
det_final_accounting.drop_duplicates(inplace=True)
det_final_accounting.head(3)

/Users/mariapalafox/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


aaref_x       pos_ID
0        C  Q3ZCM7_C354
7        C  Q3ZCM7_C303
14       C  Q3ZCM7_C239

In [148]:
# everlabeled C or K count
checkColumnValues(det_final_accounting, 'aaref_x')

  aaref_x  Count
0       K   8868
1       C   6057


In [141]:
checkColumnValues(det_final_accounting, 'Cys_react_threshold')
print()
checkColumnValues(det_final_accounting, 'Cys_target_label')
print()
checkColumnValues(det_final_accounting, 'Lys_react_threshold')
print()
checkColumnValues(det_final_accounting, 'Lys_target_label')

  Cys_react_threshold  Count
0                 Low    895
1              Medium    412
2                High     94

  Cys_target_label  Count
0      panReactive   5088
1           Target    708

  Lys_react_threshold  Count
0                 Low   3401
1              Medium    660
2                High    302

  Lys_target_label  Count
0      panReactive   7840
1           Target    115


In [143]:
not_final_accounting = not_final[["aaref", "pos_ID_falseCKtarget"]]
not_final_accounting.drop_duplicates(inplace=True)
not_final_accounting.head(3)

/Users/mariapalafox/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


aaref pos_ID_falseCKtarget
0      K          Q3ZCM7_K392
7      K          Q3ZCM7_K379
14     K          Q3ZCM7_K362

In [144]:
checkColumnValues(not_final_accounting, 'aaref')

  aaref   Count
0     K  140652
1     C   34050


### FINAL ACCOUNTING of filtered/checked CK positions for .bed file:
* total uniprot IDs = 3840
* all targets found with 7 changes per codon
* only 2 CK non-targets missing and 2 aminos with incorrect codon change count

| TYPE | Cysteine | Lysine | 
|-------|--------------|-----------|
| Low | 895 | 3401 | 
| Medium | 412 | 660 |
| High | 94 | 302 |
| Panreactive | 5088 | 7804 |
| TargetSM | 708 | 115 |
| EVER-labeled | 6057 | 8868 |
| not-labeled | 34050 | 140652 |

# [MAKE BED FILES WITH hg19 COORDINATES]

In [152]:
sampledet25 = det_final[0:25]

In [153]:
sampledet25.to_csv("detected25_sample.csv")